fastlane_core/lib/fastlane_core/helper.rb

```diff
@@ -368,8 +368,30 @@ def self.hide_loading_indicator
 
     # checks if a given path is an executable file
     def self.executable?(cmd_path)
-      # no executable files on Windows, so existing is enough there
-      cmd_path && !File.directory?(cmd_path) && (File.executable?(cmd_path) || (self.windows? && File.exist?(cmd_path)))
+      if !cmd_path || File.directory?(cmd_path)
+        return false
+      end
+
+      return File.exist?(get_executable_path(cmd_path))
+    end
+
+    # returns the path of the executable with the correct extension on Windows
+    def self.get_executable_path(cmd_path)
+      if self.windows?
+        # PATHEXT contains the list of file extensions that Windows considers executable, semicolon separated.
+        # e.g. ".COM;.EXE;.BAT;.CMD"
+        exts = ENV['PATHEXT'] ? ENV['PATHEXT'].split(';') : []
+        exts << '' # Always have an empty string (= no file extension)
```

Comment:
@joshdholtz 
This is actually incorrect. Windows will [refuse to execute files without extensions](https://social.technet.microsoft.com/Forums/windows/en-US/af026b17-4759-47d6-983a-d5b903da081e/execute-file-without-file-extension).
```suggestion
```